crates/cheatnet/src/runtime_extensions/forge_runtime_extension/mod.rs

```diff
@@ -78,6 +79,15 @@ impl BufferReaderExt for BufferReader<'_> {
             _ => unreachable!("Invalid CheatTarget variant"),
         })
     }
+
+    fn read_cheat_span(&mut self) -> BufferReadResult<CheatSpan> {
+        let cheat_span_variant = self.read_felt()?.to_u8();
+        Ok(match cheat_span_variant {
+            Some(0) => CheatSpan::Indefinite,
+            Some(1) => CheatSpan::Number(self.read_felt()?.to_usize().unwrap()),
+            _ => unreachable!("Invalid CheatSpan variant"),
```

Comment:
We should bubble errors instead there now https://github.com/foundry-rs/starknet-foundry/pull/1843/files#r1524983099

Reply:
[Bubble CheatSpan parse error](https://github.com/foundry-rs/starknet-foundry/pull/1874/commits/c30c00326d08fd72c3abf3a6bbd71d905348e439)